### PR TITLE
refactor(TopShards): migrate to ts

### PR DESCRIPTION
**Pull request by ariser**

### CODE REVIEW

File: src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx [hunk 1]
```diff
@@ -209,7 +205,6 @@ function TopShards({
                     columns={tableColumns}
                     data={data}
                     settings={TABLE_SETTINGS}
-                    className={b('table')}
```
Comment (ariser) [inline src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx:212]: it doesn't apply to DataTable

File: src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx [hunk 2]
```diff
@@ -66,126 +69,119 @@ function stringToQuerySortOrder(value) {
...
-                // eslint-disable-next-line
                 render: ({value}) => {
                     return (
                         <span
                             // tenant name is substringed out in sql query but is needed here
                             onClick={onSchemaClick(path + value)}
                             className={bLink({view: 'normal'})}
                         >
-                            {value}
+                            {value as string}
```
Comment (ariser) [inline src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx:151]: better to have a data shape, but we don't have it yet, probably will introduce it in the next PRs

File: src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx [hunk 3]
```diff
@@ -66,126 +69,119 @@ function stringToQuerySortOrder(value) {
     );
 }
 
-function dataTableToStringSortOrder(value = []) {
+function dataTableToStringSortOrder(value: SortOrder | SortOrder[] = []) {
+    if (!Array.isArray(value)) {
+        value = [value];
```
Comment (artemmufazalov) [inline src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx:74]: Assignment to function parameter doesn't look great. And it causes warning from eslint

File: src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx [hunk 4]
```diff
@@ -66,126 +69,119 @@ function stringToQuerySortOrder(value) {
...
         // which would cause an excess state change, to avoid repeating requests
         setSortOrder(dataTableToStringSortOrder(newSortOrder));
     };
 
-    const tableColumns = useMemo(() => {
+    const tableColumns: Column<any>[] = useMemo(() => {
+        const onSchemaClick = (schemaPath: string) => {
+            return () => {
+                dispatch(setCurrentSchemaPath(schemaPath));
+                dispatch(getSchema({path: schemaPath}));
```
Comment (artemmufazalov) [inline src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx:136]: Do we need to do a fetch here, if schema is fetched anyway on page reload?

**Review by artemmufazalov — CHANGES_REQUESTED** [review]

File: src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx (hunk 4 above)
Comment (ariser) [reply src/containers/Tenant/Diagnostics/TopShards/TopShards.tsx:136]: Most likely we do not. But it’s another matter, I’d prefer not to sort it out in a refactoring PR

**Review by artemmufazalov — APPROVED** [review]

**Pull request merged**